python/nvtx/nvtx.py

```diff
@@ -111,8 +110,13 @@ def mark(message=None, color="blue", domain=None):
         the event is scoped. If unspecified, the event is not associated
         with a category.
     """
-    attributes = EventAttributes(message, color)
     domain = Domain(domain)
+    category_id = (
+        self.domain.get_category_id(category)
+        if category is not None
+        else None
```

Comment:
Should we have `get_category_id` handle `None` instead of having to protect it everywhere?

Reply:
That's a good suggestion, but I like this better (even if it is repetitive and ugly) because it's more explicit about how `None` is handled. If you feel strongly, I'll change.

Reply:
Nope, just a question, defer to your judgement here.

Reply:
Let's keep this for now